graphql_compiler/compiler/ir_lowering_common/location_renaming.py

```diff
@@ -43,10 +52,12 @@ def translate_potential_location(location_translations, potential_location):
         return potential_location
 
 
-def make_location_rewriter_visitor_fn(location_translations):
+def make_location_rewriter_visitor_fn(
+    location_translations: Dict[Type[BaseLocation], Type[BaseLocation]]
+) -> Callable[[Expression], Expression]:
     """Return a visitor function that is able to replace locations with equivalent locations."""
 
-    def visitor_fn(expression):
+    def visitor_fn(expression: Any) -> Any:
```

Comment:
This is also not a good type hint, even though `mypy` will accept it due to `Any` special-casing.

Here, we have the guarantee that the `expression` input type is of the same type as the return value, and they are both bounded from above by `Expression`. That means this is `ExpressionT -> ExpressionT`, and even line 57 could be updated to state that, to be even more clear.